junit-jupiter-params/src/main/java/org/junit/jupiter/params/provider/MethodArgumentsProvider.java

```diff
@@ -43,23 +44,20 @@ public void accept(MethodSource annotation) {
 		// @formatter:off
 		return Arrays.stream(this.methodNames)
 				.map(factoryMethodName -> getMethod(context, factoryMethodName))
-				.map(method -> ReflectionUtils.invokeMethod(method, testInstance))
+				.map(method -> context.getExecutableInvoker().invoke(method, testInstance))
 				.flatMap(CollectionUtils::toStream)
 				.map(MethodArgumentsProvider::toArguments);
 		// @formatter:on
 	}
 
 	private Method getMethod(ExtensionContext context, String factoryMethodName) {
-		if (StringUtils.isNotBlank(factoryMethodName)) {
-			if (factoryMethodName.contains("#")) {
-				return getMethodByFullyQualifiedName(factoryMethodName);
-			}
-			else {
-				return ReflectionUtils.getRequiredMethod(context.getRequiredTestClass(), factoryMethodName);
-			}
+		if (StringUtils.isBlank(factoryMethodName)) {
+			return getMethodByShortName(context.getRequiredTestMethod().getName(), context.getRequiredTestClass());
```

Comment:
I think in this case we need to check for `method.getParameterCount() == 0` to preserve the existing behavior. Maybe pass a `Predicate<Method> additionalCondition` to `getMethodByShortName`?

Reply:
It depends, do we want to support parameters in this case too? If so, we could apply the "search" logic by name, and exclude the actual test method (unless we don't have a way to differentiate it from the others)

But now that I think about it, it would be a problem is the test method and factory method want to declare the exact same parameters

Maybe it's simpler to not support parameters for factory method named after the test method

Reply:
I reverted to previous behavior for now in this case (i.e. only searching for method with no parameters)